bugsnag-android-core/src/main/java/com/bugsnag/android/internal/StringUtils.kt

```diff
@@ -0,0 +1,36 @@
+package com.bugsnag.android.internal
+
+internal object StringUtils {
+    fun stringTrimmedTo(maxLength: Int, str: String): String {
+        val excessCharCount = str.length - maxLength
+        return when {
+            excessCharCount == 1 -> "${str.substring(0, maxLength)}***<1> CHAR TRUNCATED***"
+            excessCharCount > 1 -> "${str.substring(0, maxLength)}***<$excessCharCount> CHARS TRUNCATED***"
+            else -> str
+        }
+    }
+
+    fun trimNullableStringValuesTo(maxStringLength: Int, map: MutableMap<String, Any?>): Pair<Int, Int> {
+        var charCount = 0
+        @Suppress("UNCHECKED_CAST")
+        val toTrim = map.filterValues { it is String && it.length > maxStringLength } as Map<String, String>
+        toTrim.forEach {
+            charCount += it.value.length - maxStringLength
+            map[it.key] = stringTrimmedTo(maxStringLength, it.value)
+        }
```

Comment:
This might be better done using the `MutableMap.Entry` objects, which allow the value to be changed without needing additional hashmap lookups?